server/src/main/java/com/thoughtworks/go/server/service/DataSharingService.java

```diff
@@ -75,9 +77,13 @@ public void initialize() throws DuplicateMetricReporting, DuplicateDataSharingSe
         }
 
         if (existingUsageStatisticsReporting == null) {
-            UsageStatisticsReporting usageStatisticsReporting = new UsageStatisticsReporting(UUID.randomUUID().toString(), new Date());
+            UsageStatisticsReporting usageStatisticsReporting = new UsageStatisticsReporting(UUID.randomUUID().toString(), dataSharingServerUrl, new Date());
             updateUsageStatisticsReporting(usageStatisticsReporting, new HttpLocalizedOperationResult());
+        } else if (!dataSharingServerUrl.equals(existingUsageStatisticsReporting.getDataSharingServerUrl())) {
```

Comment:
This is required to make sure that if `GO_DATA_SHARING_SERVER_URL` ever changes, it is updated in the DB.

If we can ensure that  `GO_DATA_SHARING_SERVER_URL` is never going to change, we can remove this part of code.

cc: @jyotisingh

Reply:
Why store the url in db? What benefit does it bring? Its just a config. If the `GO_DATA_SHARING_SERVER_URL ` is changed to something else, we should start reporting to that instead